src/django_components/component.py

```diff
@@ -1209,44 +1209,155 @@ def _validate_outputs(self, data: Any) -> None:
+        ```
+
+    - Passed as kwarg `name`:
+
+        ```django
+        {% component rows=rows headers=headers name="my_table" ... / %}
+        ```
+
+    ### Inserting into slots
+
+    If the component defined any [slots](../concepts/fundamentals/slots.md), you can
+    pass in the content to be placed inside those slots by inserting [`{% fill %}`](#fill) tags,
+    directly within the `{% component %}` tag:
+
+    ```django
+    {% component "my_table" rows=rows headers=headers ... / %}
+      {% fill "pagination" %}
+        < 1 | 2 | 3 >
+      {% endfill %}
+    {% endcomponent %}
+    ```
+
+    ### Isolating components
+
+    By default, components behave similarly to Django's
+    [`{% include %}`](https://docs.djangoproject.com/en/5.1/ref/templates/builtins/#include),
+    and the template inside the component has access to the variables defined in the outer template.
+
+    You can selectively isolate a component, using the `only` flag, so that the inner template
+    can access only the data that was explicitly passed to it:
+
+    ```django
+    {% component "name" positional_arg keyword_arg=value ... only %}
+    ```
+    """
+
+    tag = "component"
+    end_tag = "endcomponent"
+    allowed_flags = [COMP_ONLY_FLAG]
 
     def __init__(
         self,
+        # ComponentNode inputs
         name: str,
         registry: ComponentRegistry,  # noqa F811
-        nodelist: NodeList,
-        params: TagParams,
-        isolated_context: bool = False,
+        # BaseNode inputs
+        params: List[TagAttr],
+        flags: Optional[Dict[str, bool]] = None,
+        nodelist: Optional[NodeList] = None,
         node_id: Optional[str] = None,
     ) -> None:
-        super().__init__(nodelist=nodelist or NodeList(), params=params, node_id=node_id)
+        super().__init__(params=params, flags=flags, nodelist=nodelist, node_id=node_id)
 
         self.name = name
-        self.isolated_context = isolated_context
         self.registry = registry
 
-    def __repr__(self) -> str:
-        return "<ComponentNode: {}. Contents: {!r}>".format(
-            self.name,
-            getattr(self, "nodelist", None),  # 'nodelist' attribute only assigned later.
+    @classmethod
+    def parse(  # type: ignore[override]
+        cls,
+        parser: Parser,
+        token: Token,
+        registry: ComponentRegistry,  # noqa F811
+        name: str,
+        start_tag: str,
+        end_tag: str,
+    ) -> "ComponentNode":
+        # Set the component-specific start and end tags by subclassing the base node
+        subcls_name = cls.__name__ + "_" + name
+        subcls: Type[ComponentNode] = type(subcls_name, (cls,), {"tag": start_tag, "end_tag": end_tag})
+
+        # Call `BaseNode.parse()` as if with the context of subcls.
+        node: ComponentNode = super().parse.__func__(  # type: ignore[attr-defined]
+            subcls,
+            parser,
+            token,
+            registry=registry,
+            name=name,
         )
```

Comment:
I think this could be `node = super(cls, subcls).parse(parser, token, registry=registry, name=name)`, right?

Reply:
Ah, nice one! I might have been calling `super()` with the wrong order of the the classes, because I couldn't get it to work, hence the `__func__`. Fixed now :)